doc/dts/device_tree.rst

```diff
@@ -0,0 +1,208 @@
+and also adds board specific changes like gpio/pinmux assignments.  These types
+of things will vary based on the board layout and application use.
+
+Currently supported boards
+**************************
+
+Device tree is currently supported on all ARM targets.  Support for all other
+architectures is to be done in the near future.
+
+Adding support for a board
+**************************
+
+Adding device tree support for a given board requires the following:
+
+* Base architecture support
+
+  * Add architecture specific DTs directory, if not already present.
+    Example: dts/arm for ARM.
+  * Add target to dts/<ARCH>/Makefile or create Makefile if not present
+  * Add target specific device tree files for base SoC.  These should be
+    .dtsi files to be included in the board specific device tree files.
+  * Add target specific YAML files in the dts/<ARCH>/yaml directory.
+    Create the yaml directory if not present.
+
+* SoC family support
+
+  * Add one or more SoC family .dtsi files that describe the hardware
+    for a set of devices.  The file should contain all the relevant
+    nodes and base configuration that would be applicable to all boards
+    utilizing that SoC family.
+  * Add SoC family YAML files that describe the nodes present in the .dtsi file.
+
+* Board specific support
+
+  * Add a board level .dts file that includes the SoC family .dtsi files
+    and enables the nodes required for that specific board.
+  * Board .dts file should specify the SRAM and FLASH devices, if present.
+  * Add board specific YAML files, if required.  This would occur if the
+    board has additional hardware that is not covered by the SoC family
+    .dtsi/.yaml files.
+
+* Fixup files
+
+  * Fixup files contain mappings from existing Kconfig options to the actual
+    underlying DTS derived configuration #defines.  Fixup files are temporary
+    artifacts until additional DTS changes are made to make them unncessary.
+
+Adding support for device tree in drivers
+*****************************************
+
+As drivers and other source code is converted over to make use of device tree
+generated information, these drivers may require changes to match the generated
+#define information.
+
+Source Tree Hierarchy
+=====================
+
+The device tree files are located in a couple of different directories.  The
+directory split is done based on architecture, and there is also a common
+directory where architecture agnostic device tree and yaml files are located.
+
+Assuming the current working directory is the ZEPHYR_BASE, the directory
+hierarchy looks like the following::
+
+  dts/common/
+  dts/common/yaml
+  dts/<ARCH>/
+  dts/<ARCH>/yaml
+
+The common directories contain a skeleton.dtsi include file that defines the
+address and size cells.  The yaml subdirectory contains common yaml files for
+zephyr specific nodes/properties and generic device properties common across
```

Comment:
change spellings of "zephyr" to be "Zephyr" (unless it's a directory name or in some code).